lib/avro/builder/field.rb

```diff
@@ -30,8 +30,10 @@ def initialize(name:, type_name:, record:, builder:, internal: {}, options: {},
           send(key, value) if has_dsl_attribute?(key)
         end
 
-        @type = builder.lookup(type_name, required: false) ||
-          build_type(type_name, field: self, internal: internal, options: options)
+        @type = build_type(type_name,
```

Comment:
We only want to lookup custom types. The previous code was doing the lookup for all the builtin types (string, array, etc) which I don't think makes any sense. I could make this more explicit and call `build_type` if `type_name` is in a set of known, builtin types else call `builder.lookup`.